src/commonMain/kotlin/fr/acinq/lightning/payment/IncomingPaymentHandler.kt

```diff
@@ -384,6 +383,21 @@ class IncomingPaymentHandler(val nodeParams: NodeParams, val walletParams: Walle
         return actions
     }
 
+    /**
+     * Purge all expired unpaid normal payments with creation times in the given time range.
+     *
+     * @param fromCreatedAt from absolute time in milliseconds since UNIX epoch when the payment request was generated.
+     * @param toCreatedAt to absolute time in milliseconds since UNIX epoch when the payment request was generated.
+     * @return number of invoices purged
+     */
+    suspend fun purgeExpiredPayments(fromCreatedAt: Long = 0, toCreatedAt: Long = currentTimestampMillis()): Int {
```

Comment:
Applications building on top of `lightning-kmp` won't be able to call this function right now, as the payment handler is a private field in a `Peer` instance.

Can you create a new [payment event](https://github.com/ACINQ/lightning-kmp/blob/234e049f7ee93296cfa03caa0e9de7b1f2fe33ae/src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt#L41), for example `data class PurgeExpiredPayments(val fromCreatedAt: Long, val toCreatedAt: Long) : PaymentEvent()` and add a handler for it in the `processEvent` function?
That will let Phoenix easily interact with this feature.

Reply:
Ah ha, I didn't think of that. Should PurgeExpiredPayments notify the calling application by sending a PeerListenerEvent like this then?
`object PurgeExpiredPaymentsCompleted : PeerListenerEvent()` 

(assuming I change to return a boolean and not number of purged invoices)

Reply:
Good question, I don't know if the application is really interested about knowing that it has completed.
I think they only want to trigger this as a fire-and-forget task, but @dpad85 or @robbiehanson will know better.
I'd say that for now we don't emit an event when it completes, and we can add it if they think it's useful?

Reply:
Actually shouldn't be Boolean message, but can return the number of successfully purged expired invoices :
data class PurgeExpiredPaymentsCompleted(val count: Int) : PeerListenerEvent()

Reply:
Added `PurgeExpiredPayments` in [53a9360](https://github.com/ACINQ/lightning-kmp/pull/322/commits/53a9360cad36172ef6123ba8b592e62500a18da4) but I will hold off on a return message until hearing what @dpad85 and @robbiehanson think.

Reply:
Btw, what is the policy towards removing warnings in a file I changed that weren't caused by my changes? They could just be caused by differences in linter settings, but there were a few in Peer.kt .

Reply:
There are a few about fields that could be private, but I think those shouldn't be fixed because Phoenix actually uses those fields so it needs to have access to it (maybe there's a kotlin tag to tell the compiler that these fields really need to stay public?). Apart from those, I think it's always good to fix a few warnings when you see them, even if they're unrelated to your change ;)

Reply:
Regarding the return type of `purgeExpiredPayments`, we don't need that information on Phoenix's side. We do not even need to know it succeeded.

Also, since in Phoenix the expiration information we need is [actually stored as JSON in the `incoming_payments` database](https://github.com/ACINQ/phoenix/blob/19c5f252ef2be7e107f989d1ab034b3208a4ef79/phoenix-shared/src/commonMain/paymentsdb/fr.acinq.phoenix.db/IncomingPayments.sq#L11), we won't be able to make a single SQL query that would delete things in one go. Instead, we'll have to make one SQL query for each payments in the given range, go back to kotlin, deserizalize the json data, check the expiration status and then actually make a new SQL query to delete that payment. So we can expect this operation to be a quite slow! 

Btw, I note that in our implementation of the query for expired payments, we should first check that the `received` rows are null for better perfs.